### PR TITLE
fix build with AMQP

### PR DESCRIPTION
**Pull request by benbiti**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

CMake will be failed with settings as bellow
`
-DENABLE_CASSANDRA=OFF
-DENABLE_AMQPCPP=ON
`
...

Detailed description / Documentation draft:

...
CMake Error 
`
CMake Error at contrib/libuv/CMakeLists.txt:355 (file):
  file STRINGS file "/configure.ac" cannot be read.

CMake Error at contrib/libuv/CMakeLists.txt:360 (set_target_properties):
  set_target_properties called with incorrect number of arguments.

CMake Error: File /libuv.pc.in does not exist.
CMake Error at contrib/libuv/CMakeLists.txt:365 (configure_file):
  configure_file Problem configuring file
`

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -441,6 +437,18 @@ if (USE_BROTLI)
...
 endif()
 
+if (USE_AMQPCPP)
+    if (MAKE_STATIC_LIBRARIES)
+        set (LIBUV_LIBRARY uv_a)
+    else()
+        set (LIBUV_LIBRARY uv)
+    endif()
+
+    set (LIBUV_INCLUDE_DIR "${LIBUV_ROOT_DIR}/include")
```
[the remark this answers is not in the document]
Comment (benbiti) [reply src/CMakeLists.txt]: updated as cassandra

File: cmake/find/amqpcpp.cmake [hunk 2]
```diff
@@ -9,6 +9,14 @@ if (NOT ENABLE_AMQPCPP)
     return()
 endif()
 
+
+if (NOT EXISTS "${ClickHouse_SOURCE_DIR}/contrib/libuv")
+    message (ERROR "submodule contrib/libuv is missing. to fix try run: \n git submodule update --init --recursive")
+    message (${RECONFIGURE_MESSAGE_LEVEL} "Can't find internal libuv needed for AMQP-CPP library")
+    set (USE_AMQPCPP 0)
+    return()
+endif()
```
Comment (tavplubix) [inline cmake/find/amqpcpp.cmake:18]: Do you consider extracting this code into `cmake/find/libuv.cmake` and running `libuv.cmake` before `amqpcpp.cmake` and `cassandra.cmake`?

Comment (benbiti) [reply cmake/find/amqpcpp.cmake:18]: Add a libuv.cmake and then include libuv.cmake in  amqpcpp.cmake and cassandra.cmake。

@tavplubix

**Comment by benbiti** [comment]
merge master and update

**Pull request merged**